xbmc/FileItem.cpp

```diff
@@ -668,6 +668,11 @@ void CFileItem::Serialize(CVariant& value) const
 
   if (m_gameInfoTag)
     (*m_gameInfoTag).Serialize(value["gameInfoTag"]);
+
+  if (!m_mapProperties.empty())
+    auto& customProperties = value["customproperties"];
+    for (const auto& prop : m_mapProperties)
+      customProperties[prop.first] = prop.second;
```

Comment:
Wrong logic, not what was suggested. 
Remove the `if` which is only controlling the variable declaration and not the `for` loop. Hence loop (which is always hit)  is trying to use a variable that is only sometimes defined.

Reply:
Ah, I see. I had added that in response to the desire for a "null check" before reading the array, so I figured it was still necessary.

Removed now 👌

Reply:
Or, of that one succeeds, then a *proper* conditional makes the most sense to me.

Reply:
Looks like we need Jenkins to build this again, please?

Reply:
@DaveTBlake Well, Jenkins succeeded now, so I'll try a local build to test the changes, but I think this is probably good now.

Reply:
Please keep the `if` statement because otherwise you will always create an empty `customproperties` sub-object even if it is empty.

Reply:
Alright, I've added the conditional back, and tested in a local build. This seems to be working, both to retrieve properties that exist:
`{"jsonrpc":"2.0","id":15,"method":"Files.GetDirectory","params":{"directory":"plugin://plugin.video.themoviedb.helper/?info=trending_week&nextpage=True&type=movie","media":"video","properties":["customproperties"]}}`
```json
{
    "customproperties": {
        "container.info": "trending_week",
        "container.nextpage": "True",
        "container.type": "movie",
        "dbtype": "movie",
        "item.info": "details",
        "item.nextpage": "True",
        "item.tmdb_id": "531454",
        "item.type": "movie",
        "numitems.dbid": "0",
        "numitems.tmdb": "20",
        "tmdb_id": "531454",
        "tmdb_rating": "6.5",
        "tmdb_votes": "283",
        "widget": "TMDb Helper"
    },
    "file": "plugin://plugin.video.themoviedb.helper/?info=details&type=…",
    "filetype": "directory",
    "label": "Eurovision Song Contest: The Story of Fire Saga",
    "type": "movie"
}
```
and also holds true (not sending non-existent properties) when querying a path with no properties:
`{"jsonrpc":"2.0","id":17,"method":"Files.GetDirectory","params":{"directory":"C:\\Users\\finkleandeinhorn\\Documents","media":"files","properties":["customproperties"]}}`
```json
{
    "file": "C:\\Users\\David\\Documents\\blue-dark-yellow-abstract-arti…",
    "filetype": "file",
    "label": "blue-dark-yellow-abstract-artistic-4k-hw.jpg",
    "type": "unknown"
}
```

For me, everything seems in order here. I'm going to go ahead and squash these ridiculous commits 😅

Reply:
Seems we need Jenkins to try again since I force-pushed 😉